ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java

```diff
@@ -137,17 +137,7 @@ public void onNewSlot(Date date) throws StateTransitionException, InterruptedExc
             + "  |  "
             + nodeSlot.longValue() % Constants.GENESIS_SLOT);
 
-    synchronized (this.store.getSyncObject()) {
-      try {
-        long begin = new Date().getTime();
-        while (this.store.getUnprocessedBlocks().size() == 0
-            && new Date().getTime() - begin < Constants.SECONDS_PER_SLOT * 1000 / 2) {
-          this.store.getSyncObject().wait(Constants.SECONDS_PER_SLOT * 1000 / 2);
-        }
-      } catch (InterruptedException e) {
-        LOG.log(Level.WARN, e.toString());
-      }
-    }
+    Thread.sleep(3000);
```

Comment:
Is this a hack to avoid an asynchronous implementation? I think it's ok if it's temporary but let's add the removal of this to the backlog

Reply:
it's not a hack.  It's what all clients should be doing